src/lib/markets/graph.ts

```diff
@@ -983,3 +983,140 @@ export const getPriceAtVersion = async ({
+}: {
+  graphClient: GraphQLClient
+  address: Address
+  first: number
+  offset: number
+}) {
+  const tradeHistoryQuery = gql(`
+   query fetchTradeHistory($account: Bytes!, $first: Int!, $offset: Int!) {
+    accountPositionProcesseds(
+      where: { account: $account },
+      orderBy: toOracleVersion,
+      orderDirection: desc,
+      first: $first,
+      skip: $offset,
+    ) {
+      account,
+      market,
+      fromOracleVersion,
+      toOracleVersion,
+      accumulatedValue,
+      accumulatedPnl,
+      accumulatedFunding,
+      accumulatedInterest,
+      accumulationResult_positionFee,
+      accumulationResult_keeper,
+      update {
+        version,
+        delta,
+        side,
+        collateral,
+        priceImpactFee,
+        blockTimestamp,
+        transactionHash
+      }
+    }
+   }
+  `)
+
+  const { accountPositionProcesseds } = await graphClient.request(tradeHistoryQuery, {
+    account: address,
+    first,
+    offset,
+  })
+
+  const entities = accountPositionProcesseds.reduce(
+    (acc, entity) => {
+      const market = addressToAsset(getAddress(entity.market)) as SupportedAsset
+
+      if (!acc[market]) acc[market] = []
+      acc[market].push(entity)
+      return acc
+    },
+    {} as Record<SupportedAsset, typeof accountPositionProcesseds>,
+  )
+
+  const trades = Object.entries(entities).reduce(
+    (acc, [market, entities]) => {
+      const asset = market as SupportedAsset
+
+      const updates: TradeHistoryItem<typeof accountPositionProcesseds>[] = []
+
+      entities.forEach((entity) => {
+        if (entity.update) {
+          updates.push({
+            side: entity.update.side,
+            delta: BigOrZero(entity.update.delta),
+            collateralDelta: BigOrZero(entity.update.collateral),
+            priceImpactFee: BigOrZero(entity.update.priceImpactFee),
+            fromOracleVersion: BigInt(entity.fromOracleVersion),
+            toOracleVersion: BigInt(entity.toOracleVersion),
+            accumulatedValue: BigOrZero(entity.accumulatedValue),
+            accumulatedPnl: BigOrZero(entity.accumulatedPnl),
+            accumulatedFunding: BigOrZero(entity.accumulatedFunding),
+            accumulatedInterest: BigOrZero(entity.accumulatedInterest),
+            accumulatedPositionFee: BigOrZero(entity.accumulationResult_positionFee),
+            accumulatedKeeper: BigOrZero(entity.accumulationResult_keeper),
+            date: new Date(Number(entity.update.blockTimestamp) * 1000),
+            transactionHash: entity.update.transactionHash,
+            accountPositionProcesseds: [entity],
+          })
+        } else {
+          const update = updates[updates.length - 1]
+          update.accountPositionProcesseds.push(entity)
+        }
+      })
+      acc[asset] = updates
+
+      return acc
+    },
+    {} as Record<SupportedAsset, TradeHistoryItem<typeof accountPositionProcesseds>[]>,
+  )
+  return trades
+}
+
+// Note:
+// Do i filter this one out at the end?
```

Comment:
We should treat an entity where the entity.update has 0 delta and 0 collateralDelta the same as when entity.update === null